trin-state/src/jsonrpc.rs

```diff
@@ -45,6 +53,30 @@ impl StateRequestHandler {
                         };
                     let _ = request.resp.send(response);
                 }
+                StateEndpoint::StoreContent => {
+                    let response =
+                        match StoreContentParams::<StateContentKey>::try_from(request.params) {
+                            Ok(params) => {
+                                let content_key = params.content_key.clone();
+                                let content = params.content.clone();
+                                match self
+                                    .network
+                                    .overlay
+                                    .storage
+                                    .lock()
+                                    .unwrap()
+                                    .store(&content_key, &content)
```

Comment:
Related to my comment above. First, we do not check whether the content key corresponds to data that falls within the local node's radius. Second, we don't do any validation on the content that we are requested to store for a particular content key. This is a concern because some peer could request that we store junk data for some content key, and if some other peer later offers us the valid data, then we would reject the offer because we already store the junk data.

Reply:
I'm not sure this is a concern? In order for a peer to request that we store junk data, they would need rpc access which isn't the case. I'm also not so keen on validation for proximity to the content, mostly because this is very much a dev / utility endpoint, and keeping this open and unrestricted seems like the call to me. 

I guess a concern could be if a dev adds some content to their node's database with mock data, and that in turn ends up blocking the storage of valid data from the network. Imo, I don't think this situation will occur enough to be a concern.

Reply:
I was just about to write out some further thoughts.

Not all content items defined in the portal spec contain all the information that we would need to validate them (e.g. block transaction receipts), so I'm not sure that's a viable option anyway.

I'm not sure what you mean when you say that a peer would not have RPC access.

It does seem like the radius check would limit the usefulness of this endpoint from a dev perspective.

My concern was more that invalid content may propagate beyond the local node, but like you said, it's not clear whether that would occur enough to impact the network.

Reply:
> I'm not sure what you mean when you say that a peer would not have RPC access.

In order to execute either of the `portal_*StoreContent` messages, a malicious peer would need access to a trin node's json-rpc api - which means access to send messages via ipc (where they'd need to be able to connect with the ipc file) or http (where they'd need access to the port). It doesn't seem to me that this is a reasonable concern, but I could be wrong. These endpoints are only supported via jsonrpc, it's not some new kind of message type that a malicious peer who's connected on an overlay network could use to fill a db with spam.

> My concern was more that invalid content may propagate beyond the local node, but like you said, it's not clear whether that would occur enough to impact the network.

Hmmm, yeah I'd agree with you. Firstly, in terms of quantity, this endpoint will likely be used to seed only a couple values here and there. Very valuable for testing data availability, but to test data availability only requires using a couple test datapoints, and not filling up an entire db with mock data. 

This brings up a good point, that it's not a good idea to update the `seed_database` script to use this endpoint since the script does validate the proximity of data before inserting it into the db.

I'm not sure it's even a concern that mock data will propagate? I could be wrong here, but I thought that a node will only `Offer` data to it's neighbors after receiving it via a portal network message. I don't think nodes scan their database for data and offer that to neighbors?